### PR TITLE
Cleanup selfdrive/ includes

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
Nice, I'm fine with doing this one incrementally. Feel free to mark as ready for review whenever a good chunk of it is done.

**Comment by deanlee** [comment]
format includes with google code style,sort by name, group by file type(c head file, c++head files, OP head files)

**Comment by adeebshihadeh** [comment]
Hmm, I think this makes more sense: C/C++ standard libraries, external libs, then our headers. What do you think?

**Comment by adeebshihadeh** [comment]
Also, let's leave phonelibs as is.

**Comment by deanlee** [comment]
Yes, that would be better

File: installer/installer.c [hunk 1]
```diff
@@ -1,23 +1,21 @@
-#include <stdio.h>
-#include <stdlib.h>
-#include <stdbool.h>
-#include <math.h>
-#include <unistd.h>
+#include <EGL/egl.h>
```
Comment (adeebshihadeh) [inline installer/installer.c:1]: I'd consider these system libs more like external libs than standard libs

File: installer/updater/updater.cc [hunk 2]
```diff
@@ -1,37 +1,32 @@
-#include <cstdio>
-#include <cstdlib>
-#include <cstring>
-#include <cassert>
-
-#include <unistd.h>
+#include <EGL/egl.h>
```
Comment (adeebshihadeh) [inline installer/updater/updater.cc:1]: same as above

File: selfdrive/boardd/boardd.cc [hunk 3]
```diff
@@ -1,35 +1,32 @@
-#include <stdio.h>
+#include <errno.h>
+#include <libusb-1.0/libusb.h>
```
Comment (adeebshihadeh) [inline selfdrive/boardd/boardd.cc:2]: external lib

File: selfdrive/boardd/panda.h [hunk 4]
```diff
@@ -1,13 +1,14 @@
 #pragma once
 
-#include <ctime>
-#include <cstdint>
+#include <libusb-1.0/libusb.h>
```
Comment (adeebshihadeh) [inline selfdrive/boardd/panda.h:3]: external lib

File: selfdrive/camerad/cameras/camera_common.cc [hunk 5]
```diff
@@ -1,30 +1,32 @@
-#include <thread>
-#include <chrono>
-#include <stdio.h>
+#include "selfdrive/camerad/cameras/camera_common.h"
+
 #include <assert.h>
+#include <jpeglib.h>
```
Comment (adeebshihadeh) [inline selfdrive/camerad/cameras/camera_common.cc:4]: external lib

File: selfdrive/camerad/cameras/camera_frame_stream.cc [hunk 6]
```diff
@@ -1,12 +1,12 @@
-#include "camera_frame_stream.h"
+#include "selfdrive/camerad/cameras/camera_frame_stream.h"
 
+#include <capnp/dynamic.h>
```
Comment (adeebshihadeh) [inline selfdrive/camerad/cameras/camera_frame_stream.cc:3]: external lib

File: selfdrive/ui/replay/Unlogger.cc [hunk 7]
```diff
@@ -1,21 +1,21 @@
+#include "selfdrive/ui/replay/Unlogger.h"
+
 #include <cmath>
 #include <string>
 #include <vector>
+
+// include the dynamic struct
```
Comment (adeebshihadeh) [inline selfdrive/ui/replay/Unlogger.cc:7]: doesn't make sense

File: selfdrive/ui/replay/FileReader.cc [hunk 8]
```diff
@@ -1,8 +1,9 @@
 #include "FileReader.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/replay/FileReader.cc:1]: filereader?

File: selfdrive/ui/qt/widgets/offroad_alerts.h [hunk 9]
```diff
@@ -1,13 +1,13 @@
 #pragma once
-#include <map>
 
 #include <QFrame>
-#include <QPushButton>
 #include <QLabel>
+#include <QPushButton>
 #include <QVBoxLayout>
+#include <map>
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/widgets/offroad_alerts.h:7]: stdlib above external libs

File: selfdrive/ui/qt/widgets/keyboard.h [hunk 10]
```diff
@@ -1,12 +1,11 @@
...
-#include <vector>
-
+#include <QAbstractButton>
 #include <QFrame>
+#include <QStackedLayout>
 #include <QString>
 #include <QWidget>
-#include <QStackedLayout>
-#include <QAbstractButton>
+#include <vector>
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/widgets/keyboard.h:8]: stdlib above external libs

File: selfdrive/ui/qt/spinner.h [hunk 11]
```diff
@@ -1,12 +1,11 @@
...
 #include <QLabel>
-#include <QWidget>
+#include <QOpenGLWidget>
 #include <QPixmap>
 #include <QProgressBar>
 #include <QSocketNotifier>
 #include <QVariantAnimation>
-#include <QOpenGLWidget>
+#include <QWidget>
+#include <array>
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/spinner.h:8]: stdlib above external libs

File: selfdrive/ui/qt/setup/setup.cc [hunk 12]
```diff
@@ -1,15 +1,16 @@
+#include "setup.h"
+
+#include <curl/curl.h>
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/setup/setup.cc:3]: not a stdlib

File: selfdrive/ui/qt/qt_window.h [hunk 13]
```diff
@@ -1,14 +1,13 @@
-#include <string>
-
-#include <QWidget>
 #include <QApplication>
+#include <QWidget>
+#include <string>
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/qt_window.h:3]: stdlib above external libs

File: selfdrive/ui/qt/onroad.h [hunk 14]
```diff
@@ -1,14 +1,13 @@
 #pragma once
-#include <map>
 
 #include <QSoundEffect>
 #include <QtWidgets>
+#include <map>
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/onroad.h:5]: stdlib above external libs

File: selfdrive/locationd/ublox_msg.cc [hunk 15]
```diff
@@ -1,16 +1,17 @@
-#include <cstdio>
-#include <cstdlib>
+#include "ublox_msg.h"
```
Comment (adeebshihadeh) [inline selfdrive/locationd/ublox_msg.cc:1]: full path

File: selfdrive/logcatd/logcatd_systemd.cc [hunk 16]
```diff
@@ -1,15 +1,16 @@
-#include <iostream>
+#include <systemd/sd-journal.h>
```
Comment (adeebshihadeh) [inline selfdrive/logcatd/logcatd_systemd.cc:1]: below stdlib

File: selfdrive/loggerd/tests/testraw.cc [hunk 17]
```diff
@@ -1,13 +1,13 @@
...
-#include <cstdlib>
 #include <unistd.h>
-
 #include <zmq.h>
 
-#include "common/visionipc.h"
-#include "common/timing.h"
-#include "common/util.h"
-#include "RawLogger.h"
+#include <cstdio>
```
Comment (adeebshihadeh) [inline selfdrive/loggerd/tests/testraw.cc:4]: above external libs

File: selfdrive/modeld/models/driving.cc [hunk 18]
```diff
@@ -1,14 +1,16 @@
 
-#include <string.h>
+#include "driving.h"
```
Comment (adeebshihadeh) [inline selfdrive/modeld/models/driving.cc:2]:
```suggestion
#include "selfdrive/modeld/models/driving.h"
```

File: selfdrive/hardware/eon/hardware.h [hunk 19]
```diff
@@ -1,12 +1,12 @@
 #pragma once
 
-#include <cstdlib>
-#include <fstream>
-
 #include <gui/ISurfaceComposer.h>
 #include <gui/SurfaceComposerClient.h>
 #include <hardware/hwcomposer_defs.h>
 
+#include <cstdlib>
```
Comment (adeebshihadeh) [inline selfdrive/hardware/eon/hardware.h:7]: above external libs

File: selfdrive/locationd/locationd.h [hunk 20]
```diff
@@ -1,21 +1,22 @@
 #pragma once
 
+#include <eigen3/Eigen/Dense>
```
Comment (adeebshihadeh) [inline selfdrive/locationd/locationd.h:3]: external lib

File: selfdrive/modeld/models/commonmodel.cc [hunk 21]
```diff
@@ -1,10 +1,13 @@
+#include "commonmodel.h"
```
Comment (adeebshihadeh) [inline selfdrive/modeld/models/commonmodel.cc:1]: full path

File: selfdrive/modeld/transforms/transform.cc [hunk 22]
```diff
@@ -1,9 +1,9 @@
-#include <string.h>
-#include <assert.h>
+#include "transform.h"
```
Comment (adeebshihadeh) [inline selfdrive/modeld/transforms/transform.cc:1]: full path

File: selfdrive/sensord/sensors/bmx055_gyro.cc [hunk 23]
```diff
@@ -1,8 +1,9 @@
+#include "bmx055_gyro.h"
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/bmx055_gyro.cc:1]: full path

File: selfdrive/sensord/sensors/bmx055_magn.cc [hunk 24]
```diff
@@ -1,12 +1,13 @@
-#include <cassert>
-#include <cstdio>
-#include <algorithm>
+#include "bmx055_magn.h"
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/bmx055_magn.cc:1]: full path

File: selfdrive/sensord/sensors/file_sensor.cc [hunk 25]
```diff
@@ -1,8 +1,8 @@
+#include "file_sensor.h"
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/file_sensor.cc:1]: full path

File: selfdrive/sensord/sensors/i2c_sensor.cc [hunk 26]
```diff
@@ -1,6 +1,7 @@
-#include <iostream>
 #include "i2c_sensor.h"
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/i2c_sensor.cc:1]: ful path

File: selfdrive/sensord/sensors/light_sensor.cc [hunk 27]
```diff
@@ -1,10 +1,10 @@
+#include "light_sensor.h"
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/light_sensor.cc:1]: full path

File: selfdrive/sensord/sensors/lsm6ds3_accel.cc [hunk 28]
```diff
@@ -1,9 +1,9 @@
-#include <cassert>
-#include "common/swaglog.h"
-#include "common/timing.h"
-
 #include "lsm6ds3_accel.h"
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/lsm6ds3_accel.cc:1]: full path

File: selfdrive/sensord/sensors/lsm6ds3_gyro.cc [hunk 29]
```diff
@@ -1,9 +1,10 @@
+#include "lsm6ds3_gyro.h"
```
Comment (adeebshihadeh) [inline selfdrive/sensord/sensors/lsm6ds3_gyro.cc:1]: full path

File: selfdrive/ui/paint.cc [hunk 30]
```diff
@@ -1,7 +1,9 @@
+#include "paint.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/paint.cc:1]: full path

File: selfdrive/ui/qt/offroad/networking.cc [hunk 31]
```diff
@@ -1,10 +1,11 @@
+#include "networking.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/networking.cc:1]: full path

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 32]
```diff
@@ -1,11 +1,13 @@
-#include <algorithm>
-#include <set>
+#include "wifiManager.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/offroad/wifiManager.cc:1]: full path

File: selfdrive/ui/qt/setup/setup.cc [hunk 33]
```diff
@@ -1,17 +1,20 @@
+#include "setup.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/setup/setup.cc:1]: full path

File: selfdrive/ui/qt/setup/wifi.cc [hunk 34]
```diff
@@ -1,15 +1,16 @@
+#include "wifi.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/setup/wifi.cc:1]: full path

File: selfdrive/ui/qt/spinner.cc [hunk 35]
```diff
@@ -1,16 +1,17 @@
+#include "spinner.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/spinner.cc:1]: full path

File: selfdrive/ui/qt/widgets/input.cc [hunk 36]
```diff
@@ -1,7 +1,8 @@
+#include "input.h"
```
Comment (adeebshihadeh) [inline selfdrive/ui/qt/widgets/input.cc:1]: full path

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]

**Pull request merged**